clang/lib/CConv/Constraints.cpp

```diff
@@ -35,8 +35,25 @@ Constraint::Constraint(ConstraintKind K, std::string &Rsn,
 
 // Remove the constraint from the global constraint set.
 bool Constraints::removeConstraint(Constraint *C) {
+  bool RetVal = false;
   removeReasonBasedConstraint(C);
-  return constraints.erase(C) != 0;
+  RetVal = constraints.erase(C) != 0;
+  // Delete from graph.
+  ConstraintsGraph *TG = nullptr;
+  if (Geq *GE = dyn_cast<Geq>(C)) {
+    assert(GE != nullptr && "Invalid constrains requested to be removed.");
+    // We can only remove constraints from ConstAtoms.
+    if (GE != nullptr && isa<ConstAtom>(GE->getRHS()) &&
+        isa<VarAtom>(GE->getLHS())) {
+      TG = GE->constraintIsChecked() ? ChkCG : PtrTypCG;
+      ConstraintsGraph::vertex_t SrcIdx = TG->addVertex(GE->getRHS());
+      ConstraintsGraph::vertex_t DstIdx = TG->addVertex(GE->getLHS());
+      RetVal = true;
+      // Remove the edge form the corresponding constraint graph.
+      boost::remove_edge(SrcIdx, DstIdx, TG->CG);
```

Comment:
Move as much of this as possible into the `ConstraintsGraph` class itself. At the least, add a method `removeEdge(Atom,Atom)` and put lines 49-53 in it.

Reply:
Done.